### PR TITLE
Add constructors to IdentityDbContext to allow dependency injection. …

### PR DESCRIPTION
**Pull request by danielcweber**
…The other approach (overriding 'OnConfiguring') does not work well with unit testing.

### CODE REVIEW

**Comment by HaoK** [comment]
@divega

File: src/Microsoft.AspNet.Identity.EntityFramework/IdentityDbContext.cs [hunk 1]
```diff
@@ -17,6 +18,26 @@ public class IdentityDbContext<TUser, TRole, TKey> : DbContext
...
+        {
+            
+        }
+
+        public IdentityDbContext(IServiceProvider serviceProvider, DbContextOptions options) : base(serviceProvider, options)
+        {
+            
+        }
+
+        protected IdentityDbContext()
```
Comment (divega) [inline src/Microsoft.AspNet.Identity.EntityFramework/IdentityDbContext.cs]: I think this should also do `:base()`.

Comment (divega) [inline src/Microsoft.AspNet.Identity.EntityFramework/IdentityDbContext.cs]: Actually, I am pretty sure.

Comment (HaoK) [inline src/Microsoft.AspNet.Identity.EntityFramework/IdentityDbContext.cs]: Isn't :base() implied?

Comment (HaoK) [inline src/Microsoft.AspNet.Identity.EntityFramework/IdentityDbContext.cs]: I say this because I personally have never done an explicit base() call

Comment (divega) [inline src/Microsoft.AspNet.Identity.EntityFramework/IdentityDbContext.cs]: Do'h, off course you are right.

**Comment by divega** [comment]
@HaoK what kind of test coverage do you think we should add for these constructors?

**Pull request merged**